### PR TITLE
simplify stdlib to make MintedValue a quick convert to value.

### PR DESCRIPTION
**Pull request by MicroProofs**
Also added testing methods to make use of MintedValue in unit tests.
This approach mainly focuses on converting MintedValue to Value. Thus forcing the user to deal with the knowledge of making a small quick conversion. Less type confusion as well???

### CODE REVIEW

**Comment by logicalmechanism** [comment]
I like the idea. The majority of testing I have done is either a value of zero or some token. My question about this implementation would be is it worth it to just have a public mint_value.to_value and mint_value.from_value instead?  Then we can do things like `from_value(zero())` or when values with multiple token mints or burns like `from_value(value_with_ref_and_nft)`.

**Comment by rvcas** [comment]
Hm where did all the mint specific functions go? Aesthetically I'm not exactly pleased with prefixing or suffixing mint to distinguish between the methods. Can't we just add a from_assets method to minted_value?

File: lib/aiken/transaction/value.ak [hunk 1]
```diff
@@ -311,3 +312,145 @@ test flatten_with_2() {
...
+    ]
+
+  flatten(to_value(from_list(assets))) == assets
+}
+
+/// Construct an empty `MintedValue` with nothing in it.
+/// Used mainly for unit testing purposes
+pub fn mint_zero() -> MintedValue {
+  MintedValue { inner: dict.new() }
+}
```
Comment (KtorZ) [inline lib/aiken/transaction/value.ak:436]:
```suggestion
```

File: lib/aiken/transaction/value.ak [hunk 2]
```diff
@@ -311,3 +312,145 @@ test flatten_with_2() {
...
+      |> MintedValue
+  } else {
+    let asset =
+      dict.new()
+        |> dict.insert(asset_name, quantity, bytearray.compare)
+    dict.new()
+      |> dict.insert(policy_id, asset, bytearray.compare)
+      |> MintedValue
+  }
+}
```
Comment (KtorZ) [inline lib/aiken/transaction/value.ak:456]:
```suggestion
```

**Review by KtorZ — COMMENTED** [review]
Sounds good, modulo the two new `mint_zero` and `mint_from_asset` functions. We do not want to allow constructing such values. Ever. So we shouldn't give constructors for it. The only point of the `MintedValue` is to prevent people from shooting themselves in the foot, so let's not give them footguns.

**Comment by KtorZ** [comment]
Note: we will most likely want an equivalent `from_value` function so we can still construct `MintedValue` from a sanitized `Value` (for example, to construct transactions for testing).

**Review by KtorZ — APPROVED** [review]

**Pull request merged**